cql3/query_processor.cc

```diff
@@ -1025,16 +1026,30 @@ query_processor::execute_schema_statement(const statements::schema_altering_stat
 
     cql3::cql_warnings_vec warnings;
 
+    auto request_id = guard->new_group0_state_id();
+    // TODO: either save as a member of statements::schema_altering_statement class
+    //       or extend alter_keyspace_statement::prepare_schema_mutations with guard's req id
+    const_cast<statements::schema_altering_statement&>(stmt).global_req_id = fmt::to_string(request_id);
+
     auto [ret, m, cql_warnings] = co_await stmt.prepare_schema_mutations(*this, options, guard->write_timestamp());
     warnings = std::move(cql_warnings);
 
+    ce = std::move(ret);
     if (!m.empty()) {
         auto description = format("CQL DDL statement: \"{}\"", stmt.raw_cql_statement);
-        co_await remote_.get().mm.announce(std::move(m), std::move(*guard), description);
+        if (ce && ce->target == cql_transport::event::schema_change::target_type::TABLET_KEYSPACE) {
+            co_await remote_.get().mm.announce<service::topology_change>(std::move(m), std::move(*guard), description);
+            // TODO: eliminate timeout from alter ks statement
+            auto error = co_await remote_.get().ss.wait_for_topology_request_completion(request_id);
+            if (!error.empty()) {
+                log.error("request_id: {}, error: {}", request_id, error);
```

Comment:
There's no context in this error message to figure out what failed.
Please be more verbose so the message could be useful to users/field engineering.

Reply:
I assumed that anybody using this piece of functionality will set `error` to contain all the necessary information (at least that's the case for me, hence I didn't provide any extra info, other than the req is and the error).
@kbr-scylla should I log anything extra here?

Reply:
Messages that we log, at least those above TRACE/DEBUG level, should be written in a nice human-readable English. Because humans will read those logs and wonder what the author had in mind.

"request_id: 238439875438 error: alter keyspace failed" is not nice human-readable english.

The following would be much more friendly to humans:
`CQL statement \"{stmt.raw_cql_statement}\" with topology request_id \"{request_id}\" failed with error: \"{error}\"`

I also wonder what `error` could look like.
You should try reproducing a failure and see how failures look in practice and whether you like those messages.

Reply:
OK, fixed.